src/rgw/rgw_lc.cc

```diff
@@ -585,16 +592,48 @@ static int remove_expired_obj(const DoutPrefixProvider *dpp, lc_op_ctx& oc, bool
   }
 
   obj = bucket->get_object(obj_key);
-  std::unique_ptr<rgw::sal::Object::DeleteOp> del_op = obj->get_delete_op(&oc.rctx);
-
-  del_op->params.versioning_status = obj->get_bucket()->get_info().versioning_status();
+  std::unique_ptr<rgw::sal::Object::DeleteOp> del_op
+    = obj->get_delete_op(&oc.rctx);
+  del_op->params.versioning_status
+    = obj->get_bucket()->get_info().versioning_status();
   del_op->params.obj_owner.set_id(rgw_user {meta.owner});
   del_op->params.obj_owner.set_name(meta.owner_display_name);
   del_op->params.bucket_owner.set_id(bucket_info.owner);
   del_op->params.unmod_since = meta.mtime;
   del_op->params.marker_version_id = version_id;
 
-  return del_op->delete_obj(dpp, null_yield);
+  std::unique_ptr<rgw::sal::Notification> notify
+    = store->get_notification(dpp, obj.get(), nullptr, &oc.rctx, event_type,
+			      bucket.get(), lc_id,
+			      const_cast<std::string&>(oc.bucket->get_tenant()),
+			      lc_req_id, null_yield);
+
+  /* can eliminate case when reservation is lifted into Notification */
+  auto notify_res = static_cast<rgw::sal::RadosNotification*>(notify.get())->get_reservation();
+
+  ret = rgw::notify::publish_reserve(dpp, event_type, notify_res, nullptr);
+  if (ret < 0) {
+    ldpp_dout(dpp, 1)
+      << "ERROR: notify reservation failed, deferring delete of object k="
+      << o.key
+      << dendl;
+    return ret;
+  }
+
+  ret =  del_op->delete_obj(dpp, null_yield);
+  if (ret < 0) {
+    ldpp_dout(dpp, 1) <<
+      "ERROR: publishing notification failed, with error: " << ret << dendl;
+  } else {
+      // send request to notification manager
+    ret = rgw::notify::publish_commit(
```

Comment:
dont think you want to return an error if the `publish_commit()` call fails
there delete operation is already done. i think its oik to just log

Reply:
ok, it's not fatal anyway, but I agree